packages/core/src/components/time-picker/time-picker.tsx

```diff
@@ -232,72 +655,60 @@ export class TimePicker {
-                        +inputElement.value
-                      ).toString();
-                    }}
-                  ></input>
-
-                  <ix-icon-button
-                    size="16"
-                    onClick={() =>
-                      (this._time = this._time?.minus({
-                        [descriptor.unit]: 1,
-                      }))
-                    }
-                    ghost
-                    icon={iconChevronDownSmall}
-                    variant="primary"
-                    class="arrows"
-                  ></ix-icon-button>
+                  <div class="column-header">{descriptor.header}</div>
+                  <div
+                    data-element-list-id={descriptor.unit}
+                    class="element-list"
+                    tabIndex={-1}
+                  >
+                    {descriptor.numberArray.map((number) => (
+                      <div
+                        data-element-container-id={`${descriptor.unit}-${number}`}
+                        class={{
+                          selected: this.isSelected(descriptor.unit, number),
+                          'element-container': true,
+                        }}
+                        onClick={() => this.select(descriptor.unit, number)}
+                        onFocus={() => {
+                          this.focusedUnit = descriptor.unit;
+                          this.focusedValue = number;
+                          this.onUnitCellFocus();
+                        }}
+                        onBlur={() => this.onUnitCellBlur()}
+                        tabindex={
+                          number === this.focusedValue &&
+                          descriptor.unit === this.focusedUnit
+                            ? '0'
+                            : '-1'
+                        }
+                        role="button"
+                        aria-label={`${descriptor.header}: ${number}`}
+                      >
+                        {descriptor.unit === 'millisecond'
+                          ? number.toString().padStart(3, '0')
+                          : number < 10
+                          ? `0${number}`
+                          : number}
+                      </div>
+                    ))}
+                    <div class="element-list-padding"></div>
+                  </div>
                 </div>
 
-                {index !== timepickerInformation.length - 1 && (
+                {index !== this.timePickerDescriptors.length - 1 && (
                   <div
                     class={{
                       'column-seperator': true,
                       hidden: descriptor.hidden,
                     }}
                   >
-                    :
+                    {index + 1 < this.timePickerDescriptors.length &&
+                    this.timePickerDescriptors[index + 1].unit === 'millisecond'
+                      ? '.'
+                      : ':'}
```

Comment:
Maybe we can resolve the ternary operator also before the rendering to keep template syntax more clean.